### PR TITLE
Add consensus update endpoints

### PR DESCRIPTION
**Pull request by n8maninger**
Adds consensus update endpoints for retrieving block deltas and adds experimental support for using a remote consensus database for wallet state.

### CODE REVIEW

File: wallet/manager.go [hunk 1]
```diff
@@ -327,33 +327,37 @@ func syncStore(ctx context.Context, store Store, cm ChainManager, index types.Ch
...
 
 	for _, opt := range opts {
 		opt(m)
 	}
 
+	if m.chain == nil {
+		panic("chain manager is required")
+	} else if m.store == nil {
+		panic("store is required")
+	}
```
Comment (lukechampine) [inline wallet/manager.go:347]: I don't think these need to be Options. Functions can have both mandatory and optional parameters.

File: internal/remote/chain.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+// This currently polls since the client does not expose a reorg callback yet.
+func (cm *ChainManager) OnReorg(fn func(types.ChainIndex)) (cancel func()) {
+	t := time.NewTicker(cm.refreshInterval)
+	done := make(chan struct{})
+	go func() {
+		defer t.Stop()
+		for {
+			select {
+			case <-t.C:
+				fn(cm.Tip())
```
Comment (lukechampine) [inline internal/remote/chain.go:137]: this should only call `fn` if `cm.Tip()` changes. Something like:
```go
tip := cm.Tip()
for {
    select {
    case <-t.C:
        if newTip := cm.Tip(); newTip != tip {
            tip = newTip
            fn(tip)
        }
    case <-done:
        return
    }
}
```

**Review by lukechampine — COMMENTED** [review]
This makes me wonder if, instead of doing `chain.Manager`-over-HTTP, we could do `chain.Store`-over-HTTP. Then we wouldn't need to fake or half-implement so many of these methods.

The downside is that, if you want to write a client in a different language, you have to reimplement `chain.Manager`, which is a lot of code. Hmm.

File: wallet/manager.go (hunk 1 above)
Comment (n8maninger) [reply wallet/manager.go:347]: Yeah, option-ifying all the things was a not so nice workaround to dealing with both the remote and local codepaths in the node startup. 😅 

It felt less awful than a local embedded interface
```go
var cm interface {
	wallet.ChainManager
	api.ChainManager
}
```

**Pull request closed without merging**